expression/builtin_string.go

```diff
@@ -2970,9 +2970,76 @@ func evalNumDecArgsForFormat(f builtinFunc, row chunk.Row) (string, string, bool
 		d = formatMaxDecimals
 	}
 	dStr := strconv.FormatInt(d, 10)
+	xStr, err = roundingNum(xStr, dStr)
+	if err != nil {
+		return "", "", isNull, err
+	}
 	return xStr, dStr, false, nil
 }
 
+func roundingNum(xStr, dStr string) (string, error) {
+	if strings.Contains(xStr, ".") {
+		sign := false
+		// xStr can not has '+' prefix now.
+		// It is built in `evalNumDecArgsFormat` after evaluating `Evalxxx` method.
+		if strings.HasPrefix(xStr, "-") {
+			xStr = strings.Trim(xStr, "-")
+			sign = true
+		}
+		if strings.HasPrefix(xStr, "+") {
+			return "", errors.Errorf("xStr can not has `+` prefix")
+		}
+
+		xArr := strings.Split(xStr, ".")
+		x1 := xArr[0]
+		x2 := xArr[1]
+		digit, err := strconv.Atoi(dStr)
+		if err != nil {
+			return "", err
+		}
+		if len(x2) > digit {
+			t := []byte(x2)
+			carry := false
+			if t[digit] >= '5' {
+				carry = true
+			}
+			if carry {
```

Comment:
We don't need this `if carry` check either? because the for loop already checks this condition.

Reply:
Sorry for my mistake.